src/roman.rs

```diff
@@ -183,10 +185,8 @@ impl FromStr for Roman {
     type Err = Error;
 
     fn from_str(s: &str) -> Result<Self> {
-        match RomanUnitIterator::new(s).sum::<Result<i32>>()? {
-            sum @ 1..=4999 => Ok(Roman::new_unchecked(sum as u16)),
-            sum => Err(Error::OutOfRange(sum)),
-        }
+        let sum = RomanUnitIterator::new(s).sum::<Result<u16>>()?;
```

Comment:
Again, I'm not convinced on performing this kind of math with just the `u16`. It's silly, but all it takes is `"MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM"` and we're out of range, right? =)

Reply:
I think it can be fixed by using [`checked_add`](https://doc.rust-lang.org/std/primitive.u16.html#method.checked_add), anyway this will fail in `Roman::new` in the next line.

Though crusially, `"MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM".parse::<Roman>()` will fail in `Accumulator::value` when multiplaing `66` by `1000`.

To get error here, you need something like `"MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMCM"`

Reply:
Yeah, that's what I was saying with regard to leaving the internal types for accumulator and error as larger integers: if you use the smaller integer, there's a large swath of integers that you can't really show when you're trying to say what went wrong. Obviously, this won't happen in working code, but errors aren't concerned with working code, right?

It's not a big deal, but that's what I was thinking about when I picked the big numbers, if that makes sense.